common/scripted_effects/wc_professions_reagent_effects.txt

```diff
@@ -0,0 +1,559 @@
+﻿# Character Scope
+# Calculates chance of finding reagent, then you can recall it in var:chance_current_reagent
+# REAGENT = reagent flag e.g. flag:iron or something
+# PROFESSION = profession e.g. mining or somehting
+# This is for ores
+chance_of_finding_reagent_effect = {
+	set_variable = {
+		name = chance_current_reagent
+		value = 15
+	}
+	location = {
+		save_scope_as = origin
+	}
+	change_variable = {
+		name = chance_current_reagent
+		add = {
+			# By Reagent Location
+			if = {
+				limit = { province_has_reagent_trigger = { REAGENT = this } }
+				value = 40
+			}
+			# By Rarity
+			if = {
+				limit = { reagent_rarity_is_very_common_trigger = { REAGENT = $REAGENT$ } }
+				value = 20
+			}
+			if = {
+				limit = { reagent_rarity_is_common_trigger = { REAGENT = $REAGENT$ } }
+				value = 15
+			}
+			if = {
+				limit = { reagent_rarity_is_uncommon_trigger = { REAGENT = $REAGENT$ } }
+				value = 10
+			}
+			if = {
+				limit = { reagent_rarity_is_rare_trigger = { REAGENT = $REAGENT$ } }
+				value = 5
+			}
+			if = {
```

Comment:
shouldn't these be `else_if`s? a reagent can't be rare and uncommon at the same time can it?